src/installed_dune_file.ml

```diff
@@ -59,6 +98,12 @@ let gen confs =
   in
   Sexp.List
     [ Sexp.unsafe_atom_of_string "dune"
-    ; Sexp.unsafe_atom_of_string "1"
+    ; Sexp.unsafe_atom_of_string
+        (match dune_version with
+         | (0, 0) -> "1"
+         | (1, 0) -> "2"
+         | _ ->
```

Comment:
Not sure we should be that strict. In practice, it's likely that we'll be increasing the version of the dune language often. In version 2 files, the exact version is saved in each sub-system, so we can accept any version >= 1.0 here.